src/ints/int10_modes.cpp

```diff
@@ -1953,44 +1956,210 @@ static cga_colors_t handle_cga_colors_prefs_ibm5153(const std::vector<std::strin
 	return cga_colors_ibm5153;
 }
 
-static cga_colors_t handle_cga_colors_prefs_custom(const std::vector<std::string> &cga_colors_prefs)
+std::vector<std::string> tokenize_cga_colors_pref(const std::string &cga_colors_pref)
```

Comment:
I'm struggling to understand what this function is doing.
I think it's partly because there's quite a few things named token.
 - The color format (hex vs rgb) enum is named "token"
 - but we add iterators of characters to a vector of "tokens", and not the enums.

Token is quite a generic name - it would be nice to try to eliminate it fully from in favour of closer descriptions (maybe?). 

I can atleast tell that characters from the user's color values are being put in the vector. So maybe this function is stripping the formatting/structured syntax?

I think the enum could probably be called ColorFormat or ColorType.

Reply:
I should also say that the unit tests are impressively thorough! I don't think the functional structure has to change, just some minor renaming to help new eyes like mine get up to speed a bit quicker.

Reply:
So this a pretty standard approach and naming when you write a parser, that is why I'm using the names "token" and "tokenize". First you tokenise the input and usually perform some minimal basic validation in the process, then in subsequent passes you do further validation on the resulting tokens.

So in the first step we break down the input string into tokens, which correspond to the individual colour definitions:

Input string: `#123,    #aa00ff (10, 20  , 30 )`
Tokenization results in a vector of 3 strings: `#123`, `#aa00ff`, `(10, 20  , 30)`

We need a tokenizer because the tokens can be separated by whitespaces or commas, plus the RGB-triplet definition itself can contain either spaces or commas. I guess one could write something nasty with regexes that kinda works for the happy path, but proper input validation and error handling can be only done with a proper tokeniser and parsing the input character by character.

I've renamed the `Token` enum to `TokenType`. It's not really `ColorFormat` of `ColorType`; those are just tokens,  pieces of the input that will undergo further processing. Plus we also have a `None` token, that's just a default state we don't care about between tokens (for ignoring token separator characters).

Also added some explanatory comments, let me know if this is enough in your opinion or should we try to clarify things further.

Reply:
Also, logging errors is a bit of a quick'n'dirty approach; but because this parser is only used in once place, I think it's acceptable, weighing up the pros of cons of the approach. The _really_ proper way would be to introduce error codes or at least return the error strings somehow from the parser function, not just log them as a side-effect... But that would be overengineering in this case, I think.

Reply:
Thanks for that explanation and additions to the code; makes a lot more sense now.

> Also, logging errors is a bit of a quick'n'dirty approach; but because this parser is only used in once place, I think it's acceptable, weighing up the pros of cons of the approach

I agree - the users just need a rough point that a color format went off the rails; at that point, they'll read the help messages and can compare the perfect examples of the three color formats in the conf description.